### ISSUE TITLE
Ei(-x) * exp(x) (exponential integral) limit to infinity error out

### ISSUE THREAD

**Issue by svenboulanger** [sympy/sympy#26916]
The following limit computes fine:

```
(Ei(x) * exp(-x)).limit(x, oo)
# Result: 0
```

But the following doesn't:

```
(Ei(-x) * exp(x)).limit(x, oo)
# Or equivalent: (Ei(x) * exp(-x)).limit(x, -oo)
# Expected: 0
```

Instead, sympy will error out with

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File ~\AppData\Roaming\Python\Python39\site-packages\sympy\series\gruntz.py:557, in mrv_leadterm(e, x)
    556 try:
--> 557     lt = f.leadterm(w, logx=logw)
    558 except (NotImplementedError, PoleError, ValueError):

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\core\expr.py:3533, in Expr.leadterm(self, x, logx, cdir)
   3532 if x in c.free_symbols:
-> 3533     raise ValueError(filldedent("""
   3534         cannot compute leadterm(%s, %s). The coefficient
   3535         should have been free of %s but got %s""" % (self, x, x, c)))
   3536 c = c.subs(d, log(x))

ValueError: 
cannot compute leadterm(_eis(-1/_w), _w). The coefficient should have
been free of _w but got _eis(-1/_w)

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
Cell In[84], line 2
      1 (Ei(x) * exp(-x)).limit(x, oo)
----> 2 (Ei(x) * exp(-x)).limit(x, -oo)

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\core\expr.py:3418, in Expr.limit(self, x, xlim, dir)
   3415 """ Compute limit x->xlim.
   3416 """
   3417 from sympy.series.limits import limit
-> 3418 return limit(self, x, xlim, dir)

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\series\limits.py:64, in limit(e, z, z0, dir)
     13 def limit(e, z, z0, dir="+"):
     14     """Computes the limit of ``e(z)`` at the point ``z0``.
     15 
     16     Parameters
   (...)
     61      limit_seq : returns the limit of a sequence.
     62     """
---> 64     return Limit(e, z, z0, dir).doit(deep=False)

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\series\limits.py:375, in Limit.doit(self, **hints)
    372 l = None
    374 try:
--> 375     r = gruntz(e, z, z0, dir)
    376     if r is S.NaN or l is S.NaN:
    377         raise PoleError()

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\series\gruntz.py:733, in gruntz(e, z, z0, dir)
    730     else:
    731         raise NotImplementedError("dir must be '+' or '-'")
--> 733 r = limitinf(e0, z)
    735 # This is a bit of a heuristic for nice results... we always rewrite
    736 # tractable functions in terms of familiar intractable ones.
    737 # It might be nicer to rewrite the exactly to what they were initially,
    738 # but that would take some work to implement.
    739 return r.rewrite('intractable', deep=True)

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\core\cache.py:72, in __cacheit.<locals>.func_wrapper.<locals>.wrapper(*args, **kwargs)
     69 @wraps(func)
     70 def wrapper(*args, **kwargs):
     71     try:
---> 72         retval = cfunc(*args, **kwargs)
     73     except TypeError as e:
     74         if not e.args or not e.args[0].startswith('unhashable type:'):

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\series\gruntz.py:453, in limitinf(e, x)
    451     c0, e0 = mrv_leadterm(e.min, x)
    452 else:
--> 453     c0, e0 = mrv_leadterm(e, x)
    454 sig = sign(e0, x)
    455 if sig == 1:

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\core\cache.py:72, in __cacheit.<locals>.func_wrapper.<locals>.wrapper(*args, **kwargs)
     69 @wraps(func)
     70 def wrapper(*args, **kwargs):
     71     try:
---> 72         retval = cfunc(*args, **kwargs)
     73     except TypeError as e:
     74         if not e.args or not e.args[0].startswith('unhashable type:'):

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\series\gruntz.py:563, in mrv_leadterm(e, x)
    561 incr = S.One
    562 while _series.is_Order:
--> 563     _series = f._eval_nseries(w, n=n0+incr, logx=logw)
    564     incr *= 2
    565 series = _series.expand().removeO()

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\functions\special\error_functions.py:2782, in _eis._eval_nseries(self, x, n, logx, cdir)
   2780     f = self._eval_rewrite_as_intractable(*self.args)
   2781     return f._eval_nseries(x, n, logx)
-> 2782 return super()._eval_nseries(x, n, logx)

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\core\function.py:690, in Function._eval_nseries(self, x, n, logx, cdir)
    688 a0 = [t.limit(x, 0) for t in a]
    689 if any(t.has(oo, -oo, zoo, nan) for t in a0):
--> 690     return self._eval_aseries(n, args0, x, logx)
    691 # Careful: the argument goes to oo, but only logarithmically so. We
    692 # are supposed to do a power series expansion "around the
    693 # logarithmic term". e.g.
    694 #      f(1+x+log(x))
    695 #     -> f(1+logx) + x*f'(1+logx) + O(x**2)
    696 # where 'logx' is given in the argument
    697 a = [t._eval_nseries(x, n, logx) for t in args]

File ~\AppData\Roaming\Python\Python39\site-packages\sympy\functions\special\error_functions.py:2751, in _eis._eval_aseries(self, n, args0, x, logx)
   2749 from sympy.series.order import Order
   2750 if args0[0] != S.Infinity:
-> 2751     return super(_erfs, self)._eval_aseries(n, args0, x, logx)
   2753 z = self.args[0]
   2754 l = [factorial(k) * (1/z)**(k + 1) for k in range(n)]

TypeError: super(type, obj): obj must be an instance or subtype of type
```

**Comment by oscarbenjamin** [comment]
The immediate error message results from a bug that is easily fixed:
```diff
diff --git a/sympy/functions/special/error_functions.py b/sympy/functions/special/error_functions.py
index 09279588b6..ba29c4d7d4 100644
--- a/sympy/functions/special/error_functions.py
+++ b/sympy/functions/special/error_functions.py
@@ -2748,7 +2748,7 @@ class _eis(Function):
     def _eval_aseries(self, n, args0, x, logx):
         from sympy.series.order import Order
         if args0[0] != S.Infinity:
-            return super(_erfs, self)._eval_aseries(n, args0, x, logx)
+            return super()._eval_aseries(n, args0, x, logx)
 
         z = self.args[0]
         l = [factorial(k) * (1/z)**(k + 1) for k in range(n)]
```
In this case the limit does not compute but does not exit with the error:
```python
In [1]: (Ei(-x) * exp(x)).limit(x, oo)
Out[1]: 
    ⎛ x       ⎞
lim ⎝ℯ ⋅Ei(-x)⎠
x─→∞  
```

**Comment by svenboulanger** [comment]
Great thanks! A little bit unfortunate that it can't compute the limit however. Luckily WolframAlpha didn't seem to have any issues with it.

**Comment by oscarbenjamin** [comment]
The basic bug is easy to fix.

Evaluating the limit would require having an asymptotic expansion for `Ei(x)` at `-oo`:
```python
In [5]: Ei(x).series(x, oo)
Out[5]: 
⎛120   24   6    2    1        ⎛1        ⎞⎞  x
⎜─── + ── + ── + ── + ─ + 1 + O⎜──; x → ∞⎟⎟⋅ℯ 
⎜ 5     4    3    2   x        ⎜ 6       ⎟⎟   
⎝x     x    x    x             ⎝x        ⎠⎠   
──────────────────────────────────────────────
                      x                       

In [6]: Ei(x).series(x, -oo)
...
PoleError: 
Asymptotic expansion of Ei around [-oo] is not implemented.
```

**Comment by svenboulanger** [comment]
There is an asymptotic description on [Wikipedia about Exponential Integrals](https://en.wikipedia.org/wiki/Exponential_integral) which is convergent beyond all bounds in the complex plain. Would that help?

**Comment by oscarbenjamin** [comment]
It is possibly just a case of someone needing to implement the formula somewhere.

**Comment by anutosh491** [comment]
Hey @svenboulanger as the series expansion for `Ei(x)` at `oo` and `-oo` are the same 
<img width="245" alt="image" src="https://github.com/user-attachments/assets/b3f41b1f-437d-42b3-a157-455f357edf5d">

we just had to tweak the condition to address the negative infinity cases too. The fix is in #26954 and we have the limits you pointed out working too !
